### PR TITLE
Add gogs support

### PR DESCRIPTION
**Pull request by naiba**
Gogs is a painless self-hosted Git service. https://gogs.io

### CODE REVIEW

**Comment by naiba** [comment]
@joeybloggs ping

**Comment by joeybloggs** [comment]
Sorry for the late reply, been a busy day @alrs 

Thanks for the PR will try to get to it tomorrow :) it looks like some checks failed so will also need to look at that, probably just because if the gopkg.in imports.

**Comment by joeybloggs** [comment]
@naiba fair enough, I'll try and get it merged tomorrow and open a separate issue for anyone wanting to help add tests :)

Thanks again for your work on this!

**Comment by naiba** [comment]
@joeybloggs thank you. I hope to merge as soon as possible, because I need to use it.

File: bitbucket/bitbucket.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ const (
 	PullRequestDeclinedEvent       Event = "pullrequest:rejected"
 	PullRequestCommentCreatedEvent Event = "pullrequest:comment_created"
 	PullRequestCommentUpdatedEvent Event = "pullrequest:comment_updated"
-	PullRequestCommentDeletedEvent Event = "pullrequest:comment_deleted"
+	PullRequestCommentDeletedEvent Event = "pull_request:comment_deleted"
```
Comment (joeybloggs) [inline bitbucket/bitbucket.go]: I think this may have been a merge issue, but docs say it's pullrequest

File: gogs/gogs.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		mac := hmac.New(sha256.New, []byte(hook.secret))
+		mac.Write(payload)
+
+		expectedMAC := hex.EncodeToString(mac.Sum(nil))
+
+		if !hmac.Equal([]byte(signature), []byte(expectedMAC)) {
+			webhooks.DefaultLog.Error("HMAC verification failed")
+			webhooks.DefaultLog.Debug("LocalHMAC:" + expectedMAC)
+			webhooks.DefaultLog.Debug("RemoteHMAC:" + signature)
+			webhooks.DefaultLog.Debug("Secret:" + hook.secret)
```
Comment (joeybloggs) [inline gogs/gogs.go]: Logging this is a huge security risk, will need to be removed

**Comment by joeybloggs** [comment]
Thanks Will try to merge once I'm off the train

**Pull request merged**